src/VisualStudio/Core/Def/Implementation/Options/LanguageSettingsPersisterProvider.cs

```diff
@@ -38,14 +39,9 @@ internal sealed class LanguageSettingsPersisterProvider : IOptionPersisterProvid
         public async ValueTask<IOptionPersister> GetOrCreatePersisterAsync(CancellationToken cancellationToken)
         {
             if (_lazyPersister is not null)
-            {
                 return _lazyPersister;
-            }
 
-            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
-
-            var textManager = (IVsTextManager4?)await _serviceProvider.GetServiceAsync(typeof(SVsTextManager)).ConfigureAwait(true);
-            Assumes.Present(textManager);
+            var textManager = await _serviceProvider.GetServiceAsync<SVsTextManager, IVsTextManager4>().ConfigureAwait(false);
```

Comment:
blindly casted and had an Assumes.Present before.  so this can be a standard call to GetServiceAsync.